app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -300,6 +316,20 @@ public void onComplete() {}
         };
     }
 
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_removeWatched:
+                //Solution, Scorched Earth, Copy non duplicates, clear playlist, then copy back over
+                //Other options didn't work as intended, or crashed. Like deleteItem(playlist_item) crashes when called in this function.
```

Comment:
`deleteItem(playlist_item)` probably crashes because it has to be called on the main thread (not sure though).

Reply:
Converting the AsyncTask to ReactiveX.
Does this seem good?
Any tweaks or should I push the changes?

Added
`private Disposable removeWatchedWorker;`

`onOptionsItemSelected` Looks like

``` 
RemoveWatchedStreams remover = new RemoveWatchedStreams();
// It's the same class, just without extending AsyncTask.

remover.onPreExecute();
removeWatchedWorker = Flowable.just(playlistManager.getPlaylistStreams(playlistId).blockingFirst())
    .subscribeOn(Schedulers.newThread())
    .filter((@NonNull List<PlaylistStreamEntry> playlist) -> {
        remover.doInBackground(playlist);
        return true;
    }
).observeOn(AndroidSchedulers.mainThread())
    .subscribe(playlist -> {
    remover.onPostExecute();
}, (@io.reactivex.annotations.NonNull Throwable throwable) -> {
    onError(throwable);
});
```

It still uses the scorched earth method.
I still haven't figured out how to get the `deleteItem()` to not crash.

Due to the below, I think it's the better method.
With the scorched earth method.
 1. There are less calls to the `itemListAdapter`
 2. if the app crashed, goes wrong, etc, it doesn't effect anything, it only writes once everything is done, and it's quite fast.
 3. less function calls, calling `deleteItem` will call multiple functions that aren't needed until the end.
 For example `updateThumbnailUrl()` doesn't need to be called each time, only at the end.

Reply:
Switched it over to ReactiveX in commit [fb193ce](https://github.com/TeamNewPipe/NewPipe/pull/3065/commits/fb193ce3622f34b212bff6a645887f98940eb63c)

No problem! 
I really have enjoyed NewPipe, and really thank you, and everyone else who has worked hard to make NewPipe happen!